openlibrary/catalog/add_book/match.py

```diff
@@ -1,34 +1,29 @@
+import re
+import unicodedata
 import web
-from deprecated import deprecated
-from openlibrary.catalog.utils import expand_record
-from openlibrary.catalog.merge.merge_marc import editions_match as threshold_match
-
 
 threshold = 875
 
+# fields needed for matching:
+# title, subtitle, isbn, publish_country, lccn, publishers, publish_date, number_of_pages, authors
 
-def db_name(a):
-    date = None
-    if a.birth_date or a.death_date:
-        date = a.get('birth_date', '') + '-' + a.get('death_date', '')
-    elif a.date:
-        date = a.date
-    return ' '.join([a['name'], date]) if date else a['name']
-
+re_amazon_title_paren = re.compile(r'^(.*) \([^)]+?\)$')
+re_brackets = re.compile(r'^(.+)\[.*?\]$')
+# re_brace = re.compile('{[^{}]+?}')
+re_normalize = re.compile('[^[:alpha:] ]', re.I)
```

Comment:
On closer inspection, this code is nonsense. Python doesn't support Posix character classes (perhaps this use to use `regex` instead of `re`?). I can't find anything that matches this regex, so the `re_normalize.sub()` is a no-op.

Reply:
\W is roughly equivalent, maybe we should use that instead?

Reply:
Perhaps something like [\W\s] would match the original intent, but I think the whole approach needs to be reevaluated.

Reply:
I have removed this `re_normalize` regex from the code, the only effect of the line it was used in was the associated `.strip()` cleaning whitespace from the end of the string.